From 9230af879f87354f51855adfc5e813d6bb94481b Mon Sep 17 00:00:00 2001
From: Huaming <jianghuaming.jhm@linux.alibaba.com>
Date: Sun, 1 Sep 2024 00:09:53 +0800
Subject: [PATCH 187/400] dtb:lipi:enable VI module config

1.add CMA and reserve memory for VI framebuffer and DSP
2.add VI(ISP\CSI\RY\DW200\DEC400|VIPRE|DSP) device node and enable them
3.add VI Pipeline with ov5693 sensor

Signed-off-by: Huaming <jianghuaming.jhm@linux.alibaba.com>
Signed-off-by: Han Gao <gaohan@iscas.ac.cn>
---
 .../boot/dts/thead/th1520-lichee-pi-4a.dts    | 312 +++++++++++++++++-
 arch/riscv/boot/dts/thead/th1520.dtsi         | 278 +++++++++++++++-
 2 files changed, 583 insertions(+), 7 deletions(-)

diff --git a/arch/riscv/boot/dts/thead/th1520-lichee-pi-4a.dts b/arch/riscv/boot/dts/thead/th1520-lichee-pi-4a.dts
index 7a08288665b4..098dbeb91e12 100644
--- a/arch/riscv/boot/dts/thead/th1520-lichee-pi-4a.dts
+++ b/arch/riscv/boot/dts/thead/th1520-lichee-pi-4a.dts
@@ -36,6 +36,9 @@ aliases {
 		i2s0 = &i2s0;
 		i2s1 = &i2s1;
 		i2s2 = &i2s2;
+		vivcam3 = &vvcam_sensor3;
+		viv_video2 =  &video2;
+		viv_video3 =  &video3;
 	};
 
 	chosen {
@@ -309,6 +312,210 @@ dev-thermal {
 			sustainable-power = <3000>;
 		};
 	};
+
+	soc {
+		compatible = "simple-bus";
+		interrupt-parent = <&plic>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		dma-noncoherent;
+		ranges;
+
+
+		vvcam_sensor3: vvcam_sensor@3 {//cam3 csi0 modified
+			sensor_name = "OV5693";
+			compatible = "xuantie,th1520-vvcam-sensor";
+			sensor_pdn = <&gpio1 28 0>; //powerdown pin / shutdown pin
+			sensor_rst = <&ioexp2 1 0>;
+			sensor_pdn_delay_us = <4000>; //powerdown pin / shutdown pin activated till I2C ready
+			sensor_regulators = "DOVDD18", "DVDD12", "AVDD28";
+			sensor_regulator_timing_us = <70 50 20>;
+			DOVDD18-supply = <&reg_cam0_dovdd18>;
+			DVDD12-supply = <&reg_cam0_dvdd12>;
+			AVDD28-supply = <&reg_cam0_avdd28>;
+			i2c_reg_width = /bits/ 8 <2>;
+			i2c_data_width = /bits/ 8 <1>;
+			i2c_addr = /bits/ 8 <0x36>;
+			i2c_bus = /bits/ 8 <0>;
+			status = "okay";
+		};
+
+		video2: cam_dev@300 {
+			compatible = "xuantie,th1520-video";
+			vi_mem_pool_region = <0>;	// vi_mem: framebuffer, region[0]
+			status = "okay";
+			channel0 {
+				channel_id = <0>;
+				status = "okay";
+				sensor1 {
+					subdev_name = "vivcam";
+					idx = <3>;
+					csi_idx = <0>;
+					mode_idx = <1>;
+					path_type = "SENSOR_2592x1944_LINER";
+				};
+				dma {
+					subdev_name = "vipre";
+					idx = <0>;
+					path_type = "VIPRE_CSI0_ISP1";
+				};
+				isp {
+					subdev_name = "isp";
+					idx = <1>;
+					path_type = "ISP_MI_PATH_MP";
+					output {
+						max_width = <2600>;
+						max_height = <2000>;
+						bit_per_pixel = <12>;
+						frame_count = <3>;
+					};
+				};
+			};
+			channel1 {
+				sensor1 {
+					subdev_name = "vivcam";
+					idx = <3>;
+					csi_idx = <0>;
+					mode_idx = <1>;
+					path_type = "SENSOR_2592x1944_LINER";
+				};
+				isp {
+					subdev_name = "isp";
+					idx = <1>;
+					path_type = "ISP_MI_PATH_SP";
+					output {
+						max_width = <2600>;
+						max_height = <2000>;
+						bit_per_pixel = <12>;
+						frame_count = <3>;
+					};
+				};
+			};
+			channel2 {
+				sensor1 {
+					subdev_name = "vivcam";
+					idx = <3>;
+					csi_idx = <0>;
+					mode_idx = <1>;
+					path_type = "SENSOR_2592x1944_LINER";
+				};
+				isp {
+					subdev_name = "isp";
+					idx = <1>;
+					path_type = "ISP_MI_PATH_SP2_BP";
+					output {
+						max_width = <2600>;
+						max_height = <2000>;
+						bit_per_pixel = <12>;
+						frame_count = <3>;
+					};
+				};
+			};
+		};
+
+		video3: cam_dev@400{
+			compatible = "xuantie,th1520-video";
+			vi_mem_pool_region = <0>;	// vi_mem: framebuffer, region[0]
+			status = "okay";
+			channel0 {
+				sensor1 {
+					subdev_name = "vivcam";
+					idx = <3>;
+					csi_idx = <0>;
+					mode_idx = <1>;
+					path_type = "SENSOR_2592x1944_LINER";
+				};
+				dma {
+					subdev_name = "vipre";
+					idx = <0>;
+					path_type = "VIPRE_CSI0_ISP1";
+
+				};
+				isp {
+					subdev_name = "isp";
+					idx = <1>;
+					path_type = "ISP_MI_PATH_MP";
+					output {
+						max_width = <2600>;
+						max_height = <2000>;
+						bit_per_pixel = <12>;
+						frame_count = <3>;
+					};
+				};
+				dw {
+					subdev_name = "dw";
+					idx = <0>;
+					path_type = "DW_DWE_VSE0";
+					dw_dst_depth = <2>;
+				};
+			};
+			channel1 {
+				sensor1 {
+					subdev_name = "vivcam";
+					idx = <3>;
+					csi_idx = <0>;
+					mode_idx = <1>;
+					path_type = "SENSOR_2592x1944_LINER";
+				};
+				dma {
+					subdev_name = "vipre";
+					idx = <0>;
+					path_type = "VIPRE_CSI0_ISP1";
+
+				};
+				isp {
+					subdev_name = "isp";
+					idx = <1>;
+					path_type = "ISP_MI_PATH_MP";
+					output {
+						max_width = <2600>;
+						max_height = <2000>;
+						bit_per_pixel = <12>;
+						frame_count = <3>;
+					};
+				};
+				dw {
+					subdev_name = "dw";
+					idx = <0>;
+					path_type = "DW_DWE_VSE1";
+					dw_dst_depth = <2>;
+				};
+			};
+			channel2 {
+				sensor1 {
+					subdev_name = "vivcam";
+					idx = <3>;
+					csi_idx = <0>;
+					mode_idx = <1>;
+					path_type = "SENSOR_2592x1944_LINER";
+				};
+				dma {
+					subdev_name = "vipre";
+					idx = <0>;
+					path_type = "VIPRE_CSI0_ISP1";
+
+				};
+				isp {
+					subdev_name = "isp";
+					idx = <1>;
+					path_type = "ISP_MI_PATH_MP";
+					output {
+						max_width = <2600>;
+						max_height = <2000>;
+						bit_per_pixel = <12>;
+						frame_count = <3>;
+					};
+				};
+				dw {
+					subdev_name = "dw";
+					idx = <0>;
+					path_type = "DW_DWE_VSE2";
+					dw_dst_depth = <2>;
+				};
+			};
+		};
+	};
+
 };
 
 &aon_suspend_ctrl {
@@ -802,21 +1009,48 @@ &resmem {
 	#address-cells = <2>;
 	#size-cells = <2>;
 	ranges;
-    audio_text_mem: memory@32000000 {
+
+	/* global autoconfigured region for contiguous allocations */
+	cmamem: linux,cma {
+		compatible = "shared-dma-pool";
+		reusable;
+		size = <0 0x20000000>; // 512MB on lpi4a (SOM)
+		alloc-ranges = <0 0xd8000000 0 0x20000000>; // [0x0D800_0000 ~ 0x0F800_0000]			linux,cma-default;
+		linux,cma-default;
+	};
+	dsp0_mem: memory@20000000 {             /**0x2000_0000~0x2040_0000 4M**/
+		reg = <0x0 0x20000000 0x0 0x00280000   /* DSP FW code&data section 2.5M*/
+			0x0 0x20280000 0x0 0x00001000   /* DSP communication area 4K*/
+			0x0 0x20281000 0x0 0x00007000  /* Panic/log page 28K */
+			0x0 0x20288000 0x0 0x00178000>;  /* DSP shared memory 1.5M-32K*/
+		};
+	dsp1_mem: memory@20400000 {             /**0x2040_0000~0x2080_0000 4M**/
+		reg = <0x0 0x20400000 0x0 0x00280000  /* DSP FW code&data section */
+			0x0 0x20680000 0x0 0x00001000 /* DSP communication area */
+			0x0 0x20681000 0x0 0x00007000    /* Panic/log page*/
+			0x0 0x20688000 0x0 0x00178000>;  /* DSP shared memory */
+		};
+	vi_mem: framebuffer@10000000 {
+		reg = <0x0 0x10000000 0x0 0x6700000>;	/* vi_mem_pool_region[0]  44 MB (default) */
+			//0x0 0x12C00000 0x0 0x01D00000	/* vi_mem_pool_region[1]  29 MB */
+			//0x0 0x14900000 0x0 0x01E00000>;	/* vi_mem_pool_region[2]  30 MB */
+		};
+
+	audio_text_mem: memory@32000000 {
 		reg = <0x0 0x32000000 0x0 0xE00000>;
 		//no-map;
-    };
-    audio_data_mem: memory@32E00000 {
+	};
+	audio_data_mem: memory@32E00000 {
 		reg = <0x0 0x32E00000 0x0 0x600000>;
 		//no-map;
-    };
+	};
 	audio_log_mem: memory@33400000 {
         reg = <0x0 0x33400000 0x0 0x200000>;
 	};
 	//Note: with "no-map" reserv mem not saved in hibernation
-    audio_mem: memory@32000000 {
+	audio_mem: memory@32000000 {
 		reg = <0x0 0x32000000 0x0 0x6400000>;
-    };
+	};
 	rpmsgmem: memory@1E000000 {
 		reg = <0x0 0x1E000000 0x0 0x10000>;
 	};
@@ -944,3 +1178,69 @@ &dsi0 {
 &dpu_enc0 {
 	status = "okay";
 };
+
+&isp0 {
+	status = "okay";
+};
+
+&isp1 {
+	status = "okay";
+};
+
+&isp_ry0 {
+	status = "okay";
+};
+
+&dewarp {
+	status = "okay";
+};
+
+&dec400_isp0 {
+	status = "okay";
+};
+
+&dec400_isp1 {
+	status = "okay";
+};
+
+&dec400_isp2 {
+	status = "okay";
+};
+
+&bm_visys {
+	status = "okay";
+};
+
+&bm_csi0 {
+	status = "okay";
+};
+
+&bm_csi1 {
+	status = "okay";
+};
+
+&bm_csi2 {
+	status = "okay";
+};
+
+&vidmem {
+	status = "okay";
+	memory-region = <&vi_mem>;
+};
+
+&vi_pre {
+	status = "okay";
+};
+&xtensa_dsp {
+	status = "okay";
+};
+
+&xtensa_dsp0 {
+	status = "okay";
+	memory-region = <&dsp0_mem>;
+};
+
+&xtensa_dsp1 {
+	status = "okay";
+	memory-region = <&dsp1_mem>;
+};
diff --git a/arch/riscv/boot/dts/thead/th1520.dtsi b/arch/riscv/boot/dts/thead/th1520.dtsi
index 25b6feab723d..f24f3bd8b270 100644
--- a/arch/riscv/boot/dts/thead/th1520.dtsi
+++ b/arch/riscv/boot/dts/thead/th1520.dtsi
@@ -941,8 +941,18 @@ uart5: serial@fff7f0c000 {
 		i2c0: i2c@ffe7f20000 {
 			compatible = "snps,designware-i2c";
 			reg = <0xff 0xe7f20000 0x0 0x1000>;
-			clocks = <&clk CLKGEN_I2C0_PCLK>;
 			interrupts = <44 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&clk CLKGEN_I2C0_PCLK>;
+			clock-names = "pclk";
+			clock-frequency = <100000>;
+			ss_hcnt = /bits/ 16 <0x104>;
+			ss_lcnt = /bits/ 16 <0xec>;
+			fs_hcnt = /bits/ 16 <0x37>;
+			fs_lcnt = /bits/ 16 <0x42>;
+			fp_hcnt = /bits/ 16 <0x14>;
+			fp_lcnt = /bits/ 16 <0x1a>;
+			hs_hcnt = /bits/ 16 <0x9>;
+			hs_lcnt = /bits/ 16 <0x11>;
 			#address-cells = <1>;
 			#size-cells = <0>;
 			status = "disabled";
@@ -1817,5 +1827,271 @@ vidmem: vidmem@ffecc08000 {
 			reg = <0xff 0xecc08000 0x0 0x1000>;
 			status = "okay";
 		};
+
+		bm_visys: bm_visys@ffe4040000 {
+			compatible = "xuantie,th1520-bm-visys";
+			reg = <0xff 0xe4040000 0x0 0x1000>;
+			status = "disabled";
+		};
+
+		bm_csi0: csi@ffe4000000{ //CSI2
+			compatible = "xuantie,th1520-bm-csi";
+			reg = < 0xff 0xe4000000 0x0 0x10000>;
+			interrupt-parent = <&plic>;
+			interrupts = <128 IRQ_TYPE_LEVEL_HIGH>;
+			dphyglueiftester = <0x180>;
+			sysreg_mipi_csi_ctrl = <0x140>;
+			clocks = <&visys_clk_gate TH1520_CLKGEN_MIPI_CSI0_PCLK>,
+					<&visys_clk_gate TH1520_CLKGEN_MIPI_CSI0_PIXCLK>,
+					<&visys_clk_gate TH1520_CLKGEN_MIPI_CSI0_CFG_CLK>;
+			clock-names = "pclk", "pixclk", "cfg_clk";
+			phy_name = "CSI_4LANE";
+			status = "disabled";
+		};
+
+		csia_reg: visys-reg@ffe4020000 {
+			compatible = "xuantie,th1520-visys-reg", "syscon";
+			reg = < 0xff 0xe4020000 0x0 0x10000>;
+			status = "okay";
+		};
+
+		csib_reg: visys-reg@ffe4010000{
+			compatible = "xuantie,th1520-visys-reg", "syscon";
+			reg = < 0xff 0xe4010000 0x0 0x10000>;
+			status = "okay";
+		};
+
+		bm_csi1: csi@ffe4010000{ //CSI2X2_B
+			compatible = "xuantie,th1520-bm-csi";
+			reg = < 0xff 0xe4010000 0x0 0x10000>;
+			interrupt-parent = <&plic>;
+			interrupts = <126 IRQ_TYPE_LEVEL_HIGH>; // 110 + 16 int_mipi_csi2x2_int0
+			dphyglueiftester = <0x182>; // for FPGA PHY only. ASIC not needed.
+			sysreg_mipi_csi_ctrl = <0x148>;
+			visys-regmap = <&visys_reg>;
+			csia-regmap = <&csia_reg>;
+			clocks = <&visys_clk_gate TH1520_CLKGEN_MIPI_CSI1_PCLK>,
+					<&visys_clk_gate TH1520_CLKGEN_MIPI_CSI1_PIXCLK>,
+					<&visys_clk_gate TH1520_CLKGEN_MIPI_CSI1_CFG_CLK>;
+			clock-names = "pclk", "pixclk", "cfg_clk";
+			phy_name = "CSI_B";
+			status = "disabled";
+		};
+
+		bm_csi2: csi@ffe4020000{ //CSI2X2_A
+			compatible = "xuantie,th1520-bm-csi";
+			reg = < 0xff 0xe4020000 0x0 0x10000>;
+			interrupt-parent = <&plic>;
+			interrupts = <127 IRQ_TYPE_LEVEL_HIGH>;
+			dphyglueiftester = <0x184>;
+			sysreg_mipi_csi_ctrl = <0x144>;
+			sysreg_mipi_csi_fifo_ctrl = <0x14c>;
+			csib-regmap = <&csib_reg>;
+			clocks = <&visys_clk_gate TH1520_CLKGEN_MIPI_CSI2_PCLK>,
+					<&visys_clk_gate TH1520_CLKGEN_MIPI_CSI2_PIXCLK>,
+					<&visys_clk_gate TH1520_CLKGEN_MIPI_CSI2_CFG_CLK>,
+					<&visys_clk_gate TH1520_CLKGEN_MIPI_CSI1_PCLK>,
+					<&visys_clk_gate TH1520_CLKGEN_MIPI_CSI1_PIXCLK>,
+					<&visys_clk_gate TH1520_CLKGEN_MIPI_CSI1_CFG_CLK>;
+			clock-names = "pclk", "pixclk", "cfg_clk", "pclk1", "pixclk1", "cfg_clk1";
+			phy_name = "CSI_A";
+			status = "disabled";
+		};
+
+		isp0: isp@ffe4100000 {
+			compatible = "xuantie,th1520-isp";
+			reg = <0xff 0xe4100000 0x0 0x10000>;
+			interrupt-parent = <&plic>;
+			interrupts = <117 IRQ_TYPE_LEVEL_HIGH>,<118 IRQ_TYPE_LEVEL_HIGH>;
+			clocks =  <&visys_clk_gate TH1520_CLKGEN_ISP0_ACLK>,
+					<&visys_clk_gate TH1520_CLKGEN_ISP0_HCLK>,
+					<&visys_clk_gate TH1520_CLKGEN_ISP0_PIXELCLK>,
+					<&visys_clk_gate TH1520_CLKGEN_ISP0_CLK>;
+			clock-names = "aclk", "hclk", "isp0_pclk", "cclk";
+			status = "disabled";
+		};
+
+		isp1: isp@ffe4110000 {
+			compatible = "xuantie,th1520-isp";
+			reg = <0xff 0xe4110000 0x0 0x10000>;
+			interrupt-parent = <&plic>;
+			interrupts = <120 IRQ_TYPE_LEVEL_HIGH>,<121 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&visys_clk_gate TH1520_CLKGEN_ISP0_ACLK>,
+					<&visys_clk_gate TH1520_CLKGEN_ISP0_HCLK>,
+					<&visys_clk_gate TH1520_CLKGEN_ISP0_PIXELCLK>,
+					<&visys_clk_gate TH1520_CLKGEN_ISP1_CLK>,
+					<&visys_clk_gate TH1520_CLKGEN_ISP1_PIXELCLK>;
+			clock-names = "aclk", "hclk", "isp0_pclk", "cclk", "isp1_pclk";
+			status = "disabled";
+		};
+
+		isp_ry0: isp_ry@ffe4120000 {
+			compatible = "xuantie,th1520-isp_ry";
+			reg = <0xff 0xe4120000 0x0 0x10000>;
+			interrupt-parent = <&plic>;
+			interrupts = <123 IRQ_TYPE_LEVEL_HIGH>,<124 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&visys_clk_gate TH1520_CLKGEN_ISP_RY_ACLK>,
+					<&visys_clk_gate TH1520_CLKGEN_ISP_RY_HCLK>,
+					<&visys_clk_gate TH1520_CLKGEN_ISP_RY_CCLK>;
+			clock-names = "aclk", "hclk", "cclk";
+			status = "disabled";
+		};
+
+		dewarp: dewarp@ffe4130000 {
+			compatible = "xuantie,th1520-dewarp";
+			reg = <0xff 0xe4130000 0x0 0x10000>;
+			interrupt-parent = <&plic>;
+			interrupts = <98 IRQ_TYPE_LEVEL_HIGH>,<99 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&visys_clk_gate TH1520_CLKGEN_DW200_ACLK>,
+					<&visys_clk_gate TH1520_CLKGEN_DW200_HCLK>,
+					<&visys_clk_gate TH1520_CLKGEN_DW200_CLK_VSE>,
+					<&visys_clk_gate TH1520_CLKGEN_DW200_CLK_DWE>;
+			clock-names = "aclk", "hclk", "vseclk", "dweclk";
+			status = "disabled";
+		};
+
+		dec400_isp0: dec400@ffe4060000 {
+			compatible = "xuantie,th1520-dec400";
+			reg = <0xff 0xe4060000 0x0 0x8000>;
+			status = "disabled";
+		};
+
+		dec400_isp1: dec400@ffe4068000 {
+			compatible = "xuantie,th1520-dec400";
+			reg = <0xff 0xe4068000 0x0 0x8000>;
+			status = "disabled";
+		};
+
+		dec400_isp2: dec400@ffe4070000 {
+			compatible = "xuantie,th1520-dec400";
+			reg = <0xff 0xe4070000 0x0 0x8000>;
+			status = "disabled";
+		};
+
+
+		vi_pre: vi_pre@ffe4030000 {
+			compatible = "xuantie,th1520-vi_pre";
+			reg = <0xff 0xe4030000 0x0 0x1000>;
+			interrupt-parent = <&plic>;
+			interrupts = <134 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&visys_clk_gate TH1520_CLKGEN_VIPRE_ACLK>,
+					<&visys_clk_gate TH1520_CLKGEN_VIPRE_PCLK>,
+					<&visys_clk_gate TH1520_CLKGEN_VIPRE_PIXELCLK>;
+			clock-names ="aclk", "pclk", "pixclk";
+			status = "disabled";
+		};
+
+		xtensa_dsp: dsp@01{
+			compatible = "xuantie,dsp-hw-common";
+			reg = <0xff 0xef040000 0x0 0x001000 >;   /*DSP_SYSREG(0x0000-0xFFF) */
+			status = "disabled";
+		};
+
+		xtensa_dsp0: dsp@0 {
+			compatible = "cdns,xrp-hw-simple";
+			reg = <0xff 0xe4040190 0x0 0x000010   /* host irq DSP->CPU INT Register  */
+					0xff 0xe40401e0 0x0 0x000010   /* device irq CPU->DSP INT Register  */
+					0xff 0xef048000 0x0 0x008000  /* DSP shared memory */
+					0xff 0xe0180000 0x0 0x040000>; /* DSP TCM*/
+			dsp = <0>;
+			dspsys-rst-bit = <8>;           /*bit# in DSP_SYSREG*/
+			dspsys-bus-offset = <0x90>;   /*in DSP_SYSREG*/
+			device-irq = <0x4 1 24>; /*0xff 0xe40401e4 offset to clear DSP I]RQ, bit#, IRQ# */
+			device-irq-host-offset = <0x8>;  /*0xff 0xe40401e8 offset to trigger DSP IRQ*/
+			device-irq-mode = <1>; /*level trigger*/
+			host-irq = <0x4 1>; /*0xff 0xe4040194 offset to clear, bit# */
+			host-irq-mode = <1>;   /*level trigger */
+			host-irq-offset = <0x8>;  /* 0xff 0xe4040198 offset to trigger ,device side*/
+			interrupt-parent = <&plic>;
+			interrupts = <156 IRQ_TYPE_LEVEL_HIGH>;
+			#cooling-cells = <2>;
+			firmware-name = "xrp0.elf";
+			clocks = <&dspsys_clk_gate CLKGEN_DSP0_PCLK>,
+			<&dspsys_clk_gate CLKGEN_DSP0_CCLK>;
+			clock-names = "pclk", "cclk";
+			status = "disabled";
+			operating-points-v2 = <&dsp_opp_table>;
+			dynamic-power-coefficient = <1000>;
+			#address-cells = <2>;
+			#size-cells = <1>;
+			ranges = <0x00 0x00000000 0x00 0x00000000 0xe0180000
+					0x00 0xe01c0000 0x00 0xe01c0000 0x19E40000
+					0x00 0xfa000000 0xff 0xe0000000 0x00180000
+					0x00 0xe0180000 0xff 0xe0180000 0x00040000
+					0x00 0xffc00000 0xff 0xe4000000 0x00200000 >;  /* VISYS_R */
+			dsp@0 {
+				ranges = <0x00 0x00000000 0x00 0x00000000 0xe0180000
+						0x00 0xe01c0000 0x00 0xe01c0000 0x19E40000
+						0x00 0xfa000000 0xff 0xe0000000 0x00180000
+						0x00 0xe0180000 0xff 0xe0180000 0x00040000
+						0x00 0xffc00000 0xff 0xe4000000 0x00200000 >;  /* VISYS_R */
+			};
+		};
+
+		xtensa_dsp1: dsp@1 {
+			compatible = "cdns,xrp-hw-simple";
+			reg = <0xff 0xe40401a0 0x0 0x000010   /* host irq DSP->CPU INT Register  */
+				0xff 0xe40401d0 0x0 0x000010   /* device irq CPU->DSP INT Register  */
+				0xff 0xef050000 0x0 0x008000  /* DSP shared memory */
+				0xff 0xe01C0000 0x0 0x040000>;/* DSP TCM*/
+			dsp = <1>;
+			dspsys-rst-bit = <8>;           /*bit# in DSP_SYSREG*/
+			dspsys-bus-offset = <0x90>;   /*in DSP_SYSREG*/
+			device-irq = <0x4 1 24>; /*0xff 0xe40401e4 offset to clear DSP I]RQ, bit#, IRQ# */
+			device-irq-host-offset = <0x8>;  /*0xff 0xe40401e8 offset to trigger DSP IRQ*/
+			device-irq-mode = <1>; /*level trigger*/
+			host-irq = <0x4 1>; /*0xff 0xe4040194 offset to clear, bit# */
+			host-irq-mode = <1>;   /*level trigger */
+			host-irq-offset = <0x8>;  /* 0xff 0xe4040198 offset to trigger ,device side*/
+			interrupt-parent = <&plic>;
+			interrupts = <157 IRQ_TYPE_LEVEL_HIGH>;
+			firmware-name = "xrp1.elf";
+			#cooling-cells = <2>;
+			clocks = <&dspsys_clk_gate CLKGEN_DSP1_PCLK>,
+			<&dspsys_clk_gate CLKGEN_DSP1_CCLK>;
+			clock-names = "pclk", "cclk";
+			status = "disabled";
+			operating-points-v2 = <&dsp_opp_table>;
+			dynamic-power-coefficient = <1000>;
+			#address-cells = <2>;
+			#size-cells = <1>;
+			ranges = <0x00 0x00000000 0x00 0x00000000 0xe0180000
+					0x00 0xe01c0000 0x00 0xe01c0000 0x19E40000
+					0x00 0xfa000000 0xff 0xe0000000 0x00180000
+					0x00 0xe0180000 0xff 0xe01C0000 0x00040000
+					0x00 0xffc00000 0xff 0xe4000000 0x00200000 >;  /* VISYS_R */
+			dsp@0 {
+				ranges = <0x00 0x00000000 0x00 0x00000000 0xe0180000
+						0x00 0xe01c0000 0x00 0xe01c0000 0x19E40000
+						0x00 0xfa000000 0xff 0xe0000000 0x00180000
+						0x00 0xe0180000 0xff 0xe01C0000 0x00040000
+						0x00 0xffc00000 0xff 0xe4000000 0x00200000 >;  /* VISYS_R */
+			};
+		};
+
+
+		dsp_opp_table: dsp_opp_table {
+			compatible = "operating-points-v2";
+
+			opp-125000000 {
+				opp-hz = /bits/ 64 <125000000>;
+				opp-microvolt = <800000>;
+			};
+
+			opp-250000000 {
+				opp-hz = /bits/ 64 <250000000>;
+				opp-microvolt = <800000>;
+			};
+
+			opp-500000000 {
+				opp-hz = /bits/ 64 <500000000>;
+				opp-microvolt = <800000>;
+			};
+			opp-1000000000 {
+				opp-hz = /bits/ 64 <1000000000>;
+				opp-microvolt = <800000>;
+				opp-suspend;
+			};
+		};
 	};
 };
-- 
2.43.0

